qcfractal/interface/collections/dataset.py

```diff
@@ -1171,8 +1174,19 @@ def units(self):
 
     @units.setter
     def units(self, value):
-
-        self.df *= constants.conversion_factor(self._units, value)
+        for column in self.df.columns:
+            try:
+                self.df[column] *= constants.conversion_factor(self._column_metadata[column]["units"], value)
+                if self._column_metadata[column]["units"] != self._units:
```

Comment:
there might be a better way to check this to not trigger on "kcal/mol" != "kcal / mol"

Reply:
```python
>>> qcel.constants.Quantity('kcal/mol') == qcel.constants.Quantity('kcal / mol')
True
```

Reply:
Or more concrete:
```python
>>> qcel.constants.Quantity('kcal/mol') == qcel.constants.Quantity('kilocal / mole')
True
```